src/Modal/ModalPopup.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FocusOn } from 'react-focus-on';
+import Portal from './Portal';
+import PopperElement from './PopperElement';
+import { ModalContextProvider } from './ModalContext';
+
+const ModalPopup = ({
+  children, close, isOpen, positionRef, isBlocking, withPortal, placement, ...popperProps
+}) => {
+  const RootComponent = withPortal ? Portal : React.Fragment;
+
+  return (
+    <ModalContextProvider close={close} isOpen={isOpen} isBlocking={isBlocking}>
+      <RootComponent>
+        <PopperElement target={positionRef} placement={placement} {...popperProps}>
+          <FocusOn
+            scrollLock={false}
+            enabled={isOpen}
+            onEscapeKey={close}
+            onClickOutside={close}
+          >
+            {isOpen && (
+              <div>
```

Comment:
tiny nit: do we need to render this `div` or would a React.Fragment (`<>`) work here?

Reply:
looks like fragment works fine 👍